### PR TITLE
Add torch.nn.init to list of overridable functions

### PR DESCRIPTION
**Pull request by ezyang**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #76014

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

### CODE REVIEW

File: torch/overrides.py [hunk 1]
```diff
@@ -181,6 +181,20 @@ def get_ignored_functions() -> Set[Callable]:
...
+        torch.nn.init.normal,
+        torch.nn.init.constant,
+        torch.nn.init.eye,
+        torch.nn.init.dirac,
+        torch.nn.init.xavier_uniform,
+        torch.nn.init.xavier_normal,
+        torch.nn.init.kaiming_uniform,
+        torch.nn.init.kaiming_normal,
+        torch.nn.init.orthogonal,
+        torch.nn.init.sparse,
```
[the remark this answers is not in the document]
Comment (zou3519) [reply torch/overrides.py:197]: Did you mean to add the non-deprecated underscore variants instead? e.g. torch.nn.init.uniform_

[the remark this answers is not in the document]
Comment (ezyang) [reply torch/overrides.py:197]: Check the function: this is the ignored list ;)

File: torch/overrides.py [hunk 2]
```diff
@@ -818,6 +832,12 @@ def get_testing_overrides() -> Dict[Callable, Callable]:
                                                                 distance_function=None, margin=1.0,
                                                                 swap=False, reduction='mean': -1),
         torch.nn.functional.unfold: lambda input, kernel_size, dilation=1, padding=0, stride=1: -1,
+        torch.nn.init.uniform_: lambda tensor, a=0., b=1.: -1,
+        torch.nn.init.constant_: lambda tensor, val: -1,
+        torch.nn.init.normal_: lambda tensor, mean=0., std=1.: -1,
+        torch.nn.init.constant_: lambda tensor, val: -1,
+        torch.nn.init.dirac_: lambda tensor, groups: -1,
```
Comment (zou3519) [inline torch/overrides.py:839]: dirac_ doesn't seem to have a has_torch_function? https://github.com/pytorch/pytorch/blob/cc56fac2138c8a6504bea58a3f26d2d17b187c13/torch/nn/init.py#L256-L266

**Review by zou3519 — APPROVED** [review]
I don't know what is going on with dirac_ but otherwise lgtm

File: torch/overrides.py (hunk 2 above)
Comment (ezyang) [reply torch/overrides.py:839]: looks wrong :)

**Pull request closed without merging**